### PR TITLE
Fix MIoU metric when `self.total_union==0`

### PR DESCRIPTION
**Pull request by abhi-mosaic**
@mvpatel2000 Could you try this branch out with DeepLabV3 and let me know if it fixes the issue you were seeing?

### CODE REVIEW

File: composer/metrics/metrics.py [hunk 1]
```diff
@@ -55,7 +55,9 @@ def update(self, logits: Tensor, targets: Tensor):
 
     def compute(self):
         """Aggregate state across all processes and compute final metric."""
-        return 100 * (self.total_intersect / self.total_union).mean()  #type: ignore (third-party)
+        per_class_miou = self.total_intersect / self.total_union  # type: ignore (third-party)
+        per_class_miou[self.total_union == 0] = 0
+        return 100 * per_class_miou.mean()
```
Comment (mvpatel2000) [inline composer/metrics/metrics.py:60]: Should we be excluding classes that aren't present instead of 0ing it out? CC: @Landanjs @coryMosaicML

**Comment by mvpatel2000** [comment]
LGTM, fixes NaNs ([runs](https://wandb.ai/mosaic-ml/mihir-dice?workspace=user-mihirmosaic)). Will defer to Cory or Landan for approval

File: composer/metrics/metrics.py (hunk 1 above)
Comment (coryMosaicML) [reply composer/metrics/metrics.py:60]: Better to exclude, I think. What do you think @Landanjs?

**Review by coryMosaicML — APPROVED** [review]

File: composer/metrics/metrics.py (hunk 1 above)
Comment (Landanjs) [reply composer/metrics/metrics.py:60]: I think excluding makes more sense as well. Otherwise, the mIoU will be very small depending on the number of classes and how many are present.

**Comment by Landanjs** [comment]
I will try running this, thank you for fixing my bug!

**Review by Landanjs — APPROVED** [review]
LGTM!

**Review by mvpatel2000 — APPROVED** [review]
